lib/hanami/application.rb

```diff
@@ -19,10 +19,11 @@ class Application
     class << self
       def inherited(klass)
         super
+
         @_mutex.synchronize do
           klass.class_eval do
             @_mutex = Mutex.new
-            @_configuration = Hanami::Configuration.new(application_name: name, env: Hanami.env)
+            @_configuration = Hanami::Configuration.new(application_name: klass.name, env: Hanami.env)
```

Comment:
AFAICT this should've always been like this, since we want to pass in the name of the subclass, not just `Hanami::Application` itself.